src/Chainweb/Payload.hs

```diff
@@ -774,18 +775,29 @@ instance IsCasValue PayloadWithOutputs where
     {-# INLINE casKey #-}
 
 payloadWithOutputs
-    :: PayloadData
+    :: HasCallStack
+    => PayloadData
     -> CoinbaseOutput
     -> V.Vector TransactionOutput
     -> PayloadWithOutputs
-payloadWithOutputs d co outputs = PayloadWithOutputs
-    { _payloadWithOutputsTransactions = V.zip (_payloadDataTransactions d) outputs
-    , _payloadWithOutputsMiner = _payloadDataMiner d
-    , _payloadWithOutputsCoinbase = co
-    , _payloadWithOutputsPayloadHash = _payloadDataPayloadHash d
-    , _payloadWithOutputsTransactionsHash = _payloadDataTransactionsHash d
-    , _payloadWithOutputsOutputsHash = _payloadDataOutputsHash d
-    }
+payloadWithOutputs d co outputs =
+  if V.length (_payloadDataTransactions d) /= V.length outputs
```

Comment:
Lars do we still want this length check? The zip here wasn't an issue length-wise (just that the order was wrong)

Reply:
It's O(1) right? Doesn't hurt ...

Reply:
Yes, I think, it's cheap and good to have.

Add `HasCallStack` constraint?

Maybe prefix the message with "Internal code invariant violation: " so that it is clear that this should never happen and if it happens it is a code bug.